### PR TITLE
Update Customize-how-a-Questionnaire-is-displayed.md

### PR DESCRIPTION
**Pull request by MJ1998**
Update documentation for custom widget support.

**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #[issue number]

**Description**
Clear and concise code change description.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Builds | Releases | Other)

**Screenshots (if applicable)**

**Checklist**

- [ ] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/CODE_OF_CONDUCT.md).
- [ ] I have read the [Contributing](https://google.github.io/android-fhir/contrib/) page.
- [ ] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [ ] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [ ] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [ ] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 1]
```diff
@@ -77,55 +77,25 @@ questions. However, if you want to render a question in a way not described in
...
+
+### 2. Register the above factory with datacapture library via DataCaptureConfig
+Within your [DataCaptureConfig](https://github.com/google/android-fhir/blob/ced8527a5481972591615ad4364487e89130fb6e/datacapture/src/main/java/com/google/android/fhir/datacapture/DataCaptureConfig.kt#L35) object, assign the custom factory matchers provider to the `questionnaireItemViewHolderFactoryMatchersProviderFactory` property. This step makes your provider available for resolving view holders during questionnaire rendering. ([example](https://github.com/google/android-fhir/blob/ced8527a5481972591615ad4364487e89130fb6e/catalog/src/main/java/com/google/android/fhir/catalog/CatalogApplication.kt#L42C5-L47C8))
+
+### 3. Configure the QuestionnaireFragment Builder
+When constructing your QuestionnaireFragment using the builder pattern, utilize the `setCustomQuestionnaireItemViewHolderFactoryMatchersProvider` method. Provide the appropriate string identifier to specify which custom widgets to use. ([example](https://github.com/google/android-fhir/blob/ced8527a5481972591615ad4364487e89130fb6e/catalog/src/main/java/com/google/android/fhir/catalog/DemoQuestionnaireFragment.kt#L142C13-L150C23))
```
Comment (jingtang10) [inline docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:93]:
```suggestion
When constructing your QuestionnaireFragment using the builder pattern, call the `setCustomQuestionnaireItemViewHolderFactoryMatchersProvider` method with the appropriate string identifier to specify which custom widgets to use. ([example](https://github.com/google/android-fhir/blob/ced8527a5481972591615ad4364487e89130fb6e/catalog/src/main/java/com/google/android/fhir/catalog/DemoQuestionnaireFragment.kt#L142C13-L150C23))
```

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 2]
```diff
@@ -77,55 +77,25 @@ questions. However, if you want to render a question in a way not described in
...
+### 1. Create a factory of custom ViewHolderFactoryMatchersProvider
+Develop a class that implements the [QuestionnaireItemViewHolderFactoryMatchersProviderFactory](https://github.com/google/android-fhir/blob/ced8527a5481972591615ad4364487e89130fb6e/datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt#L563) interface. This class acts as a registry for associating custom widgets with their corresponding view holder factories. ([example](https://github.com/google/android-fhir/blob/master/catalog/src/main/java/com/google/android/fhir/catalog/ContribQuestionnaireItemViewHolderFactoryMatchersProviderFactory.kt))
+
+As part of this you need to define:-
+
+1. factory: The custom component factory to use.
+2. matches: A predicate function which, given a
+    [`Questionnaire.QuestionnaireItemComponent`](https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/Questionnaire.QuestionnaireItemComponent.html), returns true if the factory should apply to that item.
+
+### 2. Register the above factory with datacapture library via DataCaptureConfig
```
Comment (jingtang10) [inline docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:89]:
```suggestion
### 2. Register the factory with datacapture library
```

**Review by jingtang10 — CHANGES_REQUESTED** [review]
thanks jajoo for fixing our doc!

@aditya-07 made some changes to the custom widget mechanism. @aditya-07 can you also review this doc change? thanks

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 3]
```diff
@@ -77,55 +77,25 @@ questions. However, if you want to render a question in a way not described in
...
-    [`Questionnaire.QuestionnaireItemComponent`](https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-structures-r4/org/hl7/fhir/r4/model/Questionnaire.QuestionnaireItemComponent.html),
-    returns true if the factory should apply to that item.
-2. Create a custom implementation of `QuestionnaireFragment` that
-    overrides `getCustomQuestionnaireItemViewHolderFactoryMatchers`
-    ([example](https://github.com/google/android-fhir/blob/master/catalog/src/main/java/com/google/android/fhir/catalog/CustomQuestionnaireFragment.kt#L22)).
-    It should return a list that contains your
-    `QuestionnaireItemViewHolderFactoryMatcher`.
-3. When rendering your questionnaire, use your custom implementation of
-    <code>QuestionnaireFragment</code> instead.
+### 1. Create a factory of custom ViewHolderFactoryMatchersProvider
```
[the remark this answers is not in the document]
Comment (MJ1998) [reply docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:80]: Yes done in #2580

File: docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md [hunk 4]
```diff
@@ -77,55 +77,25 @@ questions. However, if you want to render a question in a way not described in
 the FHIR standard of the SDC implementation guide, you can create a custom
 component.
 
-### Create a custom component
```
[the remark this answers is not in the document]
Comment (MJ1998) [reply docs/use/SDCL/Customize-how-a-Questionnaire-is-displayed.md:80]: reverted and improved in #2580 2580

**Comment by MJ1998** [comment]
Closing this as the PR is moved to #2580

**Pull request closed without merging**